extension/src/popup/components/sendPayment/SendAmount/index.tsx

```diff
@@ -301,7 +303,29 @@ export const SendAmount = ({
   // remove non digits and decimal
   const cleanAmount = (s: string) => s.replace(/[^0-9.]/g, "");
 
-  const formatAmount = (val: string) => {
+  const preserveCursor = (
```

Comment:
I've tried to house the hairy logic into this one fn called preserveCursor.
Basically, this is my logic for tracking where the cursor should be after updates/clean/sanitize.

1. Determine wether we need to sanitize digits and/or decimals.
2. For digits only, compare the previous value with the newly formatted value and move the cursor according to the difference in number of commas, and difference in number of characters in the raw value vs the cleaned/sanitized(even though we filter out anything but numbers, invalid chars still move the cursor).
3. If digits & decimals, do previous step on chars before the dot and also account for characters after the dot that are cleaned out but have moved the cursor.

Reply:
Nice - this is a really good usability improvement. Maybe we should move `preserveCursor` and `formatAmount` to a helpers file as we'd probably want to reuse this for future currency inputs and it'd be nice to separate some of this complex logic from the tsx markup. Wdyt?

Reply:
Would also be great to add your explanation of the logic from this PR comment into a comment in the code as I found it very helpful to understand what was happening

Reply:
Yeah totally, makes sense. I'll move these out into helpers and was going to take a stab at writing tests for them since they're kind of hairy.

Reply:
This is done now, going to close this thread.